### PR TITLE
Move the MSAL desktop cache extension + tests

### LINKED ISSUE #3152: [Feature Request] Migrate MSAL repo Extensions into MSAL.NET repo
**Is your feature request related to a problem? Please describe.**
We should migrate the MSAL extensions into MSAL to help alleviate the complexity of the release pipeline.

**Describe the solution you'd like**
Migrate the MSAL extension project into MSAL.
Modify build pipeline to build/release MSAL extensions packages.

### PR DESCRIPTION
**Pull request by bgavrilMS**
~Currently no build changes.~ Builds should work.

Fixes #3152

To help reviewers - do not review any newly added file (actual lib content and tests). All this logic already exists.

### CODE REVIEW

**Comment by pmaytak** [comment]
Is this actually ready for review or more work needs to be done? What changes were made - is it just moving the the cache extensions code under the cache extensions folder (I see some other unrelated changes?)?

What build related changes need to be done? I think we have the release/ci/pr builds. Would we use OneBranch for this also?

**Comment by bgavrilMS** [comment]
> Is this actually ready for review or more work needs to be done? What changes were made - is it just moving the the cache extensions code under the cache extensions folder (I see some other unrelated changes?)?
> 
> What build related changes need to be done? I think we have the release/ci/pr builds. Would we use OneBranch for this also?

You are right, there are some build changes remaining. Moving back to draft.

**Comment by bgavrilMS** [comment]
@pmaytak - do you remember how to fix the test error "n exception occurred while invoking executor 'executor://nunit3testexecutor/': Could not load type 'System.Runtime.Remoting.Channels.IChannel' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'." ?

**Comment by pmaytak** [comment]
> @pmaytak - do you remember how to fix the test error "n exception occurred while invoking executor 'executor://nunit3testexecutor/': Could not load type 'System.Runtime.Remoting.Channels.IChannel' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'." ?

I think it was just updating to the latest version of those dependencies?
This PR maybe? https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4036

**Review by gladjohn — APPROVED** [review]

File: LibsAndSamples.sln [hunk 1]
```diff
@@ -111,18 +111,14 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "pipelines", "pipelines", "{
 		build\template-android-appcenter-tests.yaml = build\template-android-appcenter-tests.yaml
 		build\template-build-and-prep-automation.yaml = build\template-build-and-prep-automation.yaml
 		build\template-build-and-run-all-tests.yaml = build\template-build-and-run-all-tests.yaml
-		build\template-build-android-automation.yaml = build\template-build-android-automation.yaml
-		build\template-build-ios-automation.yaml = build\template-build-ios-automation.yaml
```
Comment (pmaytak) [inline LibsAndSamples.sln:115]: Seems unrelated to cache changes, move to separate PR?

File: LibsAndSamples.sln [hunk 2]
```diff
@@ -111,18 +111,14 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "pipelines", "pipelines", "{
...
 		build\template-ci-and-pr.yaml = build\template-ci-and-pr.yaml
 		build\template-desktop-unit-and-automation.yaml = build\template-desktop-unit-and-automation.yaml
 		build\template-install-dotnet-core.yaml = build\template-install-dotnet-core.yaml
 		build\template-install-keyvault-secrets.yaml = build\template-install-keyvault-secrets.yaml
 		build\template-ios-appcenter-tests.yaml = build\template-ios-appcenter-tests.yaml
 		build\template-nuget-pack.yaml = build\template-nuget-pack.yaml
-		build\template-OneBranch-CI-libsandsamples.yaml = build\template-OneBranch-CI-libsandsamples.yaml
-		build\template-OneBranch-Release-libsandsamples.yaml = build\template-OneBranch-Release-libsandsamples.yaml
-		build\template-OneBranch-releasebuild-netonly.yaml = build\template-OneBranch-releasebuild-netonly.yaml
+		build\template-OneBranch-libsandsamples.yaml = build\template-OneBranch-libsandsamples.yaml
```
[the remark this answers is not in the document]
Comment (trwalke) [reply LibsAndSamples.sln:121]: Im assuming this branch has the latest changes right?
Is this a merge error? These files should be there.

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs [hunk 3]
```diff
@@ -30,6 +30,7 @@
 namespace Microsoft.Identity.Test.Unit.PublicApiTests
 {
     [TestClass]
+    [DeploymentItem(@"Resources\valid_cert.pfx")]
```
[the remark this answers is not in the document]
Comment (trwalke) [reply tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs:33]: Yeah, sometimes I ran into this issue locally. this should resolve it

**Review by trwalke — CHANGES_REQUESTED** [review]
Missing files for OneBranch Build

**Comment by bgavrilMS** [comment]
@trwalke - what files are missing?

File: LibsAndSamples.sln (hunk 2 above)
[the remark this answers is not in the document]
Comment (bgavrilMS) [reply LibsAndSamples.sln]: I now added all 3 OneBranch files - please take a look


build\template-OneBranch-CI-libsandsamples.yaml = build\template-OneBranch-CI-libsandsamples.yaml
build\template-OneBranch-Release-libsandsamples.yaml = build\template-OneBranch-Release-libsandsamples.yaml
build\template-OneBranch-releasebuild-netonly.yaml = build\template-OneBranch-releasebuild-netonly.yaml

**Review by trwalke — APPROVED** [review]

**Pull request merged**